packages/@expo/cli/src/start/server/metro/MetroBundlerDevServer.ts

```diff
@@ -802,7 +802,7 @@ export class MetroBundlerDevServer extends BundlerDevServer {
     };
 
     // https://github.com/facebook/metro/blob/2405f2f6c37a1b641cc379b9c733b1eff0c1c2a1/packages/metro/src/lib/parseOptionsFromUrl.js#L55-L87
-    if (!opts.mainModuleName.startsWith('/')) {
+    if (!opts.mainModuleName.startsWith('/') && !path.isAbsolute(opts.mainModuleName)) {
```

Comment:
`opts.mainModuleName` can be `C:\Users\...`. without this change, the path would be `./C:\Users\...`, and metro cannot resolve the path

Reply:
I'm not sure about this, but shouldn't the `mainModuleName` be a module specifier? As in, not be an absolute path, but the path of either `serverRoot` or `projectRoot` + `mainModuleName`

Reply:
it's happening from `npx expo export:embed` and the mainModuleName would in some cases passing an absolute path like `C:\Users\...`.

i'm keeping this change and merging the pr. if there's any problems, i can follow up in a separate pr.